packages/inferno/src/DOM/utils/common.ts

```diff
@@ -96,11 +105,33 @@ export function callAllAnimationHooks(animationQueue: Function[], callback?: Fun
   // since all animations are to be run and we can't predict the order in which they complete.
   let fn;
   while ((fn = animationQueue.pop()) !== undefined) {
-    fn(() => {
-      if (--animationsLeft <= 0 && isFunction(callback)) {
-        callback();
-      }
-    });
+    try {
+      fn(() => {
+        if (--animationsLeft <= 0 && isFunction(callback)) {
+          callback();
+        }
+      });
+    } catch (e) {
```

Comment:
what was the reason to try catch and re-throw the error?

Reply:
To open for discussion. In production we might want to swallow errors there but in development we want probably want to show them.

Reply:
Same on row 126

Reply:
Hmm, but in case it fails to exception is it able to recover? 
What are the gains of hiding the exceptions. At the moment inferno does not catch or raise any exceptions, maybe it should be kept that way? User could implement try catch on application side if they see it necessary

Reply:
We need all the calls to complete in order for the dom-manipulations to be performed (move/remove). That said, I agree that hiding exceptions is a recipe for problems and there should be no case where an exception in the animation handlers is an acceptable outcome.

I would prefer to remove them, exceptions caught here should be fixed by the developer/animation library.

Reply:
Yes, I do think it will recover. Worst case is that the animation isn't performed. But I still feel it hides errors that should be fixed.

Reply:
hey, can you get latest from master branch and remove try - catches, and those utils for debugging. then I think we can merge this

Reply:
I'll fix it before Jeff Bezos makes it to space! 😁